src/bst.py

```diff
@@ -0,0 +1,300 @@
+"""Creatae a Binary Search Tree."""
+
+
+class Node:
+    """Tree node: left and right child & data which can be any object."""
+
+    def __init__(self, val=None):
+        """Initialize the BST data Structure."""
+        self.left = None
+        self.right = None
+        self.val = val
+        self.parent = None
+        self._counter = 0
+
+
+class BinarySearchTree:
+    """Instantiates a BST."""
+
+    def __init__(self):
+            """Set the root."""
+            self.root = None
+
+    def insert(self, val):
+        """Insert new node with data."""
+        if self.val:
```

Comment:
BinarySearchTree instances don't have val, left, right, or _counter attributes.  Those are on the node instances.

Reply:
Whenever you start some process on a bst, you start from the root (which will be either a node or None) and traverse left or right from there. It's helpful if you have some variable that keeps track of which node you're currently comparing against.